pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +92,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
+	AuthPlugin string `json:"auth-plugin,omitempty"`
```

Comment:
Okay. How about something like:

```
type AuthProviderConfig struct {
	Name string
	Config json.RawMessage
}
```

Plugins would then be provided with the specified Config when instantiated. Simple plugins (like mine) can specify a Name, and leave Config blank. More complex plugins could define and parse their own config, and it would all look nice and pretty in the kubeconfig file.